ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/JsonRpcConfiguration.java

```diff
@@ -102,6 +104,30 @@ public void setHostsWhitelist(final List<String> hostsWhitelist) {
     this.hostsWhitelist = hostsWhitelist;
   }
 
+  public boolean isAuthenticationEnabled() {
+    return authenticationEnabled;
+  }
+
+  public void setAuthenticationEnabled(final boolean authenticationEnabled) {
+    this.authenticationEnabled = authenticationEnabled;
+  }
+
+  public void setAuthenticationCredentialsFile(final String authenticationCredentialsFile) {
```

Comment:
Why have the JsonRpc be concerned with the manner of authentications, perhaps an abstraction around that would be a smoother approach?

Reply:
`JsonRpcConfiguration` isn't just about `JSON-RPC`. Maybe the name isn't great but it actually has got info re the 'HTTP' listener (e.g. port, host, cors config, etc.)

So this is the right place to add the credentials config, if we need to change something it would be the name (but I'd rather have this as part of a separate PR).